src/pool/srv_pool.c

```diff
@@ -2387,10 +2376,12 @@ ds_pool_connect_handler(crt_rpc_t *rpc, int handler_version)
 
 	/* Check existing pool handles. */
 	d_iov_set(&key, in->pci_op.pi_hdl, sizeof(uuid_t));
-	d_iov_set(&value, &hdl, sizeof(hdl));
+	d_iov_set(&value, NULL, 0);
 	rc = rdb_tx_lookup(&tx, &svc->ps_handles, &key, &value);
 	if (rc == 0) {
-		if (hdl.ph_flags == in->pci_flags) {
+		/* found it */
+		hdl = value.iov_buf;
+		if (hdl->ph_flags == in->pci_flags) {
```

Comment:
Since hdl points to a persistent memory address, if we go to out_map_version before the new D_ALLOC(hdl, ...) call, we will call the new D_FREE(hdl) on a persistent memory address, leading to potential memory corruptions. I would suggest introducing a new struct pool_hdl pointer, such as hdl_pmem, for this block of code.

Reply:
Oops, right and thanks to point it !!